src/core/execute.h

```diff
@@ -330,6 +330,7 @@ struct ExecContext {
         int bind_log_sockets;
         int memory_ksm;
         PrivateTmp private_tmp;
+        PrivateTmp private_var_tmp;
```

Comment:
hmm, why is this a variable of its own?

why not instead have a helper that determines this on the fly?

i.e. exec_context_private_var_tmp() that checks private_tmp, and the rootdir/rootimage stuff and returns the right value?

Reply:
i think it might be ok to keep this two separate fields in NamespaceParameters, but in ExecContext make this one field + one accessor

Reply:
I initially tried that, but it depends on Unit.default_dependencies, and mapping from ExecContext to Unit is not straight forward. Moreover, the contents of Unit is not serialized to systemd-executor. So, I decided to use a variable.

Reply:
hmm, i see the problem, but can we at least initialize private_var_tmp to _PRIVATE_TMP_INVALID then, and then in unit_patch_contexts() set it to the final value, based on private_tmp, in all cases?

i.e. i'd prefer if we can always discern the cases private_var_tmp is off, from private_var_tmp is just not patched in yet, for whatever reason, so that any issues with not initializing this for some reason are caught.

also please add a comment explaining that private_var_tmp is not really an independent option, but always initialized from private_tmp